frontend/src/actions/authActions.ts

```diff
@@ -0,0 +1,60 @@
+import {Dispatch} from 'redux'
+
+import {actionCreator} from '../redux-utils/actionCreator'
+import {Action} from '../types/action'
+import {AuthApiClient } from '../api/authRequest'
+import {IUser} from '../models/user'
+
+export enum AuthActionType {
+  LOG_OUT = 'auth/LOG_OUT',
+  SET_USER = 'auth/SET_USER'
+}
+
+export interface IUserDetails {
+  email: string
+  password: string
+}
+
+export interface IUserAuthApiResponse {
+  token: string
+  access: string
+  user: IUser
+}
+
+export type fetchUserAction = Action<AuthActionType.SET_USER, IUser>
+export type logoutUserAction = Action<AuthActionType.LOG_OUT>
+
+export type AuthActions = fetchUserAction | logoutUserAction
+
+export const setUserAction = actionCreator<AuthActionType.SET_USER, IUser>(AuthActionType.SET_USER)
+
+// TODO: localStorage.clear()
```

Comment:
When we want to implement this TODO? Please, fix this, or file a new ticket for this and left a reference in comment.

Reply:
added PA-55 for Logout to make Logout and fix LocalStorage.clear()